notes/cryptanalysis.tex

```diff
@@ -59,6 +266,97 @@ \section{Block Cipher Design and Cryptanalysis}
 \end{lemma}
 
 
-\bnm
-  \AdvPRF{\EM,P}{\advA} \le \frac{2\cdot q_1 \cdot q_2}{2^n}
-\enm
+%\bnm
+%  \AdvPRF{\EM,P}{\advA} \le \frac{2\cdot q_1 \cdot q_2}{2^n}
+%\enm
+
+
+\textbf{Recovering many bits of des} The basic intuition is to realize that we can build partial linear approximations of only a subset
+of the rounds using the piling up lemma. A combination of these linear approximations should enable us to recover multiple bits of des
+and brute force the rest. 16 round des breaks with \(2^{43} \)  known plaintext/ciphertext pairs.
+\ )
+
+Linear cryptanalysis can mainly be used to reduce the search space of keys based on the recovered bits and launch a known plaintext attack.
+
+\subsubsection*{differential cryptanalysis}
+Differential cryptanalysis is similar to its linear counterpart in that it exploits the high probability of certain occurrences of plaintext
+differences and differences into the output. However, it aims to capture the relationship using non-linear expressions.
+
+\[ \delta_c = E_k(M + \delta_m) + E_k(M) \] 
+The aim here is to find \( \delta_m \) such that \( \delta_c \) holds for the given expression with a high probability over the given choice of M. \newline
+
+In DES, Multiple of these differentials for s-boxes can be chained together like in the linear case and this lead to recovering bits of the key.
+
+\scribenote{potential question idea: difference between DES, 2DES and 3DES in terms of security}
```

Comment:
2DES, either as HW or as part of notes:
- Given a brute force attack for DES, how to build the meet-in-the-middle attack for 2DES.

3DES - class didn't go through proving 3DES security. Do we know how difficult the proof is?